When running my random tests for the outpost card I noticed every single test failed. Upon further inspection of the code, I found the outpost flag is never set. In fact the flag is set to a negative number during the tests. Specifically at the line: 692 within the function outpostCard there is the following line is the bug: state->outpostPlayed--;  instead this line should be state->outpostPlayed++;
 
The next bug I noticed while running my random tests was every single one of my tests for adventurer were failing. I looked into my test for adventurer to find out what I was initially testing for in the adventurer card and found for any player after using the card is not receiving exactly 2 cards from the dealer. This is occurring from line 653 (while(drawntreasure<1){) this causes only one treasure to be drawn from the deck, Otherwise the adventurer card looks completely fine. 

When running my unit tests for the smithy card I found that smithy is drawing an extra card then it should. 4 instead of 3. Upon exception of the code I found on line 678 within the for loop the range is between 0 and 4. I had to add a print statement to count out the iterator variable because on first glance it looked completely fine.
